src/scene/light.js

```diff
@@ -713,24 +718,23 @@ class Light {
 
     getBoundingSphere(sphere) {
         if (this._type === LIGHTTYPE_SPOT) {
-            const range = this.attenuationEnd;
+
+            // based on https://bartwronski.com/2017/04/13/cull-that-cone/
+            const size = this.attenuationEnd;
             const angle = this._outerConeAngle;
-            const f = Math.cos(angle * math.DEG_TO_RAD);
+            const cosAngle = this._outerConeAngleCos;
             const node = this._node;
+            tmpVec.copy(node.up);
 
-            spotCenter.copy(node.up);
-            spotCenter.mulScalar(-range * 0.5 * f);
-            spotCenter.add(node.getPosition());
-            sphere.center = spotCenter;
-
-            spotEndPoint.copy(node.up);
-            spotEndPoint.mulScalar(-range);
-
-            tmpVec.copy(node.right);
-            tmpVec.mulScalar(Math.sin(angle * math.DEG_TO_RAD) * range);
-            spotEndPoint.add(tmpVec);
+            if (angle > 45) {
+                tmpVec.mulScalar(-size * cosAngle);
+                sphere.radius = size * this._outerConeAngleSin;
+            } else {
+                tmpVec.mulScalar(size / (-2 * cosAngle));
+                sphere.radius = size / (2 * cosAngle);
```

Comment:
I don't know how performance critical this is, it just looks like the same calculation (negated), so that:

```suggestion
                const r = size / (2 * cosAngle);
                tmpVec.mulScalar(-r);
                sphere.radius = r;
```

(completely untested)

Reply:
yep don't see why not